org.eclipse.ice.commands/src/main/java/org/eclipse/ice/commands/FileHandlerFactory.java

```diff
@@ -45,16 +45,19 @@ public FileHandlerFactory() {
 	 * 
 	 * @return FileHandler - instance of FileHandler that does the transfer
 	 */
-	public IFileHandler getFileHandler() throws IOException {
-		IFileHandler handler = null;
+	public FileHandler getFileHandler(ConnectionConfiguration connectionConfig) throws IOException {
+		FileHandler handler = null;
 
-		// TODO - determine how to identify local vs. remote FileHandler
-		boolean isLocal = true;
+		// Determine if the hostname in the config is local or not
+		boolean isLocal = isLocal(connectionConfig.getHostname());
 
+		// If it is, return a local file handler, otherwise remote
 		if (isLocal) {
 			handler = new LocalFileHandler();
 		} else {
 			handler = new RemoteFileHandler();
+			// Set the connection information for the remote file handler
+			((RemoteFileHandler) handler).setConnectionConfiguration(connectionConfig);
```

Comment:
Type casting should be avoided and seems indicative of a problem with the model. Why are you not passing the configuration via the constructor?.

Reply:
In this particular case, just do this:
```java
RemoteFileHandler remoteHandler = new RemoteFileHandler();
remoteHandler.setConnectionConfiguration(connectionConfig);
handler = remoteHandler;
```

Reply:
Okay I see, I will change this and the other type casting instances to something similar